Review Iteration 2, group 3

anonim-no opened the pull request:
**Кол-во людей в команде:** 4

**Декомпозиция эпиков (ссылка на таск-трекер):** 
https://github.com/users/anonim-no/projects/2/

**Реализуемые задачи во втором спринте:**
 - фильтры и всё что с ними связано

**Первоначальная оценка времени выполнения задач:** 10 дней

**Реальное время, потраченное на реализацию:**  13 дней

**Список выполненных задач/требований:**
всё требования в полной версии

**Приложение (apk):**
[group_3_iteration_2.zip](https://github.com/anonim-no/practicum-android-diploma/files/13214395/group_3_iteration_2.zip)

JudjinGM committed:
Поправил баг с восстановлением состояния экрана

JudjinGM committed:
Поправил баг с восстановлением состояния экрана:
1. Переделал paginationLoadingState на SingleLiveEvent.
2. Поправил баг повторных вызовов Toast.  Добавил debounce.
3. Переписал экран Похожие вакансии, наследовался от Поиск вакансий тк было слишком много повторного кода.

anonim-no committed:
Merge pull request #202 from anonim-no/bugfix/SRCH-200/restore_state

Поправил баг с восстановлением состояния экрана

JudjinGM committed:
Добавил Firebase Crashlytics

manualcheg committed:
1. Добавлены модули зависимостей всех слоев для koin
2. SharedPreferences описана как зависимость koin
3. Создан класс- модель данных Filter и сопутствующие для хранения значений фильтров в SharedPreferences.
4. Создан класс для хранения констант экрана фильтров
5. Сделан класс конвертер для преобразования Filter в параметры поискового запроса
6. Начато создание реализации методов FilterOptionsDataSourceImpl

manualcheg committed:
1. Добавлен класс между SharedPrefs и DataSource
2. создан новый класс для хранения объекта Filter до применения фильтров к поиску и методы для добавления параметров фильтрации

manualcheg committed:
1. В интерфейс для работы с параметрами фильтрации добавлен метод очистки
2. Доделана реализация интерфейса для работы с параметрами фильтрации
3. Реализованы методы добавления значений параметров фильтрации

manualcheg committed:
1. Созданы в классе FilterOptionsDataSource функции добавления значений фильтров

manualcheg committed:
1. Созданы в классе LocalCache функции добавления значений фильтров

manualcheg committed:
Merge remote-tracking branch 'origin/feature/FLTR/prepare_for_use_SharedPrefs' into feature/FLTR/prepare_for_use_SharedPrefs

JudjinGM committed:
Убрал проверку на краш

JudjinGM committed:
добавил Leak Canary

manualcheg committed:
1. Переименованы классы
2. Добавлена очистка FilterLocalCache после полу

JudjinGM committed:
1. Перенес модели в common.
2. Добавил очистку кеша после сохранения фильтров d sharedPref.
3. Внес изменения в SarchRepository, для того чтобы проект собирался

JudjinGM committed:
Merge pull request #207 from anonim-no/feature/FLTR/prepare_for_use_SharedPrefs

Feature/fltr/prepare for use shared prefs

JudjinGM committed:
Незначительные правки

JudjinGM committed:
Реализовал поиск по фильтрам

JudjinGM committed:
Реализовал изменение кнопки фильтра

JudjinGM committed:
Поправил ЗП на зарплату

anonim-no committed:
Merge pull request #209 from anonim-no/feature/TECH-206/firebase_crashlytics

Добавил Firebase Crashlytics

anonim-no committed:
Merge pull request #210 from anonim-no/feature/TECH-208/leakCanary

Добавил Leak Canary

anonim-no committed:
Merge pull request #212 from anonim-no/feature/SRCH-203/filter_button_state

Реализовал поиск по фильтрам и изменение иконки фильтров, при включеных фильтрах

anonim-no committed:
Revert "Добавил Firebase Crashlytics"

anonim-no committed:
Merge pull request #214 from anonim-no/revert-209-feature/TECH-206/firebase_crashlytics

Revert "Добавил Firebase Crashlytics"

anonim-no committed:
- объеденил region_and_sector и filter в одну фичу

BigMag74 committed:
Добавил Repository и UseCase'ы для Фильтрации

BigMag74 committed:
Добавил Зависимости в DI и viewModel

BigMag74 committed:
Добавил базовые классы для экранов Выбор места работы и Выбор страны
Добавил Юзкейсы для получения списка Стран, регионов, отраслей

BigMag74 committed:
Добавил зависимости для получения списка стран, регионов, отраслей

anonim-no committed:
- немного оптимизации экрана выбора фильтров

JudjinGM committed:
Создал DTO классы для запросов областей, стран и отраслей

JudjinGM committed:
Создал классы запросов и обертки ответов , а также добавил методы в Api для получения списков стран, отраслей и регионов

JudjinGM committed:
Переименовал FilterOptionsDataSource в FiltersLocalDataSource

JudjinGM committed:
Дополнил метод do request для получеиня списка стран, регионов и отраслей

JudjinGM committed:
Дополнил метод do request для получения списка стран, регионов и отраслей

JudjinGM committed:
В процессе написания маппера для данных

BigMag74 committed:
Merge pull request #220 from anonim-no/feature/FLTR-218/Repository_and_useCases_for_filter_screen

Добавил Repository и UseCase'ы для Фильтрации

BigMag74 committed:
Merge branch 'dev' of https://github.com/anonim-no/practicum-android-diploma into feature/FLTR-219/DI_for_filter

anonim-no committed:
Merge pull request #216 from anonim-no/tech/FLTR-215/region_and_sector_to_filter

- объеденил region_and_sector и filter в одну фичу

BigMag74 committed:
Поправил код, согласно замечаниям
Не трогал вложенность if-else и верстку

anonim-no committed:
- общая оптимизация верстки

anonim-no committed:
- merge in dev

anonim-no committed:
- bugfix merge in dev

JudjinGM committed:
1. Добавил методы в репозиторий для получения областей, стран и отраслей
2. Создал и реализовал маппер(конвертор) dto моделей в domain модели.
   Из репозитория возращаеться области без стран, страны и отрасли.
3. Добавил методы для получения стран по id в api и dataSource
4. Переименовал модели Area на модели вакансий и модели фильтров
5. Временно написал во вьюмодель и фрагмент метод получения и вывода данных в Logcat для отладки. Перед пушем удалю.

JudjinGM committed:
Merge branch 'dev' into treeeeeee

# Conflicts:
#	app/src/main/java/ru/practicum/android/diploma/filter/data/repositoryImpl/FilterRepositoryImpl.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/domain/repository/FilterRepository.kt

anonim-no committed:
- правки по ревью

JudjinGM committed:
Поправил UseCase, Repository после merge.
Поправил ошибку в Industries тк в id приходит стринга типа 42.597

BigMag74 committed:
Merge branch 'dev' into feature/FLTR-219/DI_for_filter

BigMag74 committed:
Merge pull request #221 from anonim-no/feature/FLTR-219/DI_for_filter

Добавил Зависимости в DI и viewModel

BigMag74 committed:
Merge branch 'dev' of https://github.com/anonim-no/practicum-android-diploma into bugfix/TECH-224/first_iteration_refactoring

BigMag74 committed:
Merge pull request #226 from anonim-no/bugfix/TECH-224/first_iteration_refactoring

Поправил код, согласно замечаниям

JudjinGM committed:
Добавил поле в AreaFilter название страны, для того, чтобы упростить получение информации на экране "Регион и страна выбраны"

JudjinGM committed:
Вроде как поправил, если будут предложения, напишите, пожалуйста

anonim-no committed:
Merge branch 'dev' into refactoring/FLTR-217/layout_optimization

JudjinGM committed:
Поправил метод, убрал лишнюю вложенность

JudjinGM committed:
Merge branch 'dev' into feature/FLTR-213/areas_countries_industries_fetch

# Conflicts:
#	app/src/main/java/ru/practicum/android/diploma/filter/di/filterDomainModule.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/ui/viewModel/FilteringSettingsViewModel.kt
#	app/src/main/java/ru/practicum/android/diploma/search/data/mapper/VacancyDtoConverter.kt
#	app/src/main/java/ru/practicum/android/diploma/search/data/repositoryImpl/SearchRepositoryImpl.kt
#	app/src/main/java/ru/practicum/android/diploma/search/ui/fragment/SearchFragment.kt

JudjinGM committed:
Merge pull request #229 from anonim-no/feature/FLTR-213/areas_countries_industries_fetch

Изучить и заполнить NetworkApi c необходимыми запросами по экрану фильтрация

JudjinGM committed:
Preparing for merge #1

JudjinGM committed:
merge fix

anonim-no committed:
Merge pull request #228 from anonim-no/refactoring/FLTR-217/layout_optimization

Рефикторинг верстки экрана фильтров

JudjinGM committed:
Реализвал customView для экрана Настройки фильтрации

JudjinGM committed:
Merge pull request #236 from anonim-no/refactoring/SRCH-235/refactoring

Поправить замечание ревьювера касательно вложенности if/else/when

anonim-no committed:
- добавил недостающие фрагменты фильтров

anonim-no committed:
Merge pull request #241 from anonim-no/feature/FLTR-240/sector_country_fragments

- добавил недостающие фрагменты фильтров

anonim-no committed:
- добавил недостающие viewModel экрана фильтров
- переименовал Region-Area, Sector-Industry

anonim-no committed:
- попытка исправить ошибку git, удаление файла

anonim-no committed:
- попытка исправить ошибку git, восстановление файла

anonim-no committed:
Merge branch 'dev' into feature/FLTR-237/settings_custom_view

anonim-no committed:
- добавлен фрагмент и viewModel для экрана выбора места работы

anonim-no committed:
Merge pull request #244 from anonim-no/feature/FLTR-243/fragment_choosing_workplace

- добавлен фрагмент и viewModel для экрана выбора места работы

JudjinGM committed:
1. Поправил замечания.
2. Добавил mapper и FilterUi
3. Немного переименовал названия

JudjinGM committed:
Merge remote-tracking branch 'origin/feature/FLTR-237/settings_custom_view' into feature/FLTR-237/settings_custom_view

# Conflicts:
#	app/src/main/java/ru/practicum/android/diploma/filter/di/FilterUiModule.kt

JudjinGM committed:
Merge pull request #238 from anonim-no/feature/FLTR-237/settings_custom_view

Реализовал customView для экрана Настройки фильтрации

JudjinGM committed:
Поправил filterDomainModule

BigMag74 committed:
Добавил обработку состояний для полей страна и регион,
Добавил кнопку Выбрать

BigMag74 committed:
Заменил TextInputLayout на CustomView

BigMag74 committed:
Изменил обработку состояний в соответствии с CustomView

BigMag74 committed:
Добавил обработку нажатий полей Страна и регион

manualcheg committed:
1. Некое начальное состояние

manualcheg committed:
1. Чекбокс работает

manualcheg committed:
1. Зарплата работает

manualcheg committed:
1. Применение работает

manualcheg committed:
1. Работает всё, за исключением:
• очистка и обновление Места работы
• очистка и обновление Отрасли

manualcheg committed:
1. Работает всё, за исключением:
• очистка и обновление Места работы
• очистка и обновление Отрасли

2. Небольшие правки

BigMag74 committed:
Добавил переменные в навигацию и отрефакторил согласно новому подходу

BigMag74 committed:
Добавил переменные в навигацию и отрефакторил согласно новому подходу

BigMag74 committed:
Сделал AreaFilter и CountryFilter сериализуемыми, поправил навигацию

anonim-no committed:
Merge pull request #246 from anonim-no/feature/FTLR/230_new_ui_for_filter_settings_with_custom_view

Экран Настройки фильтрации с кастомным view

anonim-no committed:
Merge branch 'dev' into feature/FLTR-231/chooshing_workplace_UI

anonim-no committed:
Merge pull request #245 from anonim-no/feature/FLTR-231/chooshing_workplace_UI

Добавил обработку состояний для полей страна и регион,

anonim-no committed:
- добавлены анимации
- добавлены клики на поля место работы и отрасль

anonim-no committed:
Merge pull request #248 from anonim-no/feature/FLTR-247/animations

- добавлены анимации

JudjinGM committed:
Реализовал получение регионов по приходчщим аргументам(id страны). Если id == null, то приходит весь списко регионов. Сделал филтрацию по имени региона.

Навигация в этой ветке неверная, сделал для возможности перейти сразу на выбор регинов.

manualcheg committed:
1. Код объединённый с кодом правок Ильнура

manualcheg committed:
1. Исправление навигации

manualcheg committed:
Merge pull request #254 from anonim-no/feature/FLTR-253/create_feature_screen_final_version

Доработка экрана Фильтрация поиска

BigMag74 committed:
Добавил состояния кнопки "Выбрать" и попытался реализовать передачу параметров, однако не получилось

BigMag74 committed:
Пофиксил передачу данных между экранами

BigMag74 committed:
Реализовал UI слой экрана Выбор страны
Обработка нажатий кнопок назад, находятся тут:
https://github.com/anonim-no/practicum-android-diploma/pull/256

BigMag74 committed:
Добавил плейсхолдеры и их обработку

JudjinGM committed:
Поправил замечания:
1. Клик на весь элемент
2. Навигация и передача area через fragment result api.

manualcheg committed:
различные пробные варианты. Пока без результатов

BigMag74 committed:
Поправил замечание с адаптером

BigMag74 committed:
Merge pull request #256 from anonim-no/bugfix/FLTR-252/choosing_workplace_UI

Bugfix/fltr 252/choosing workplace UI

BigMag74 committed:
merge commit

BigMag74 committed:
Merge pull request #257 from anonim-no/feature/FLTR-232/filtering_country_UI

Реализовал UI слой экрана Выбор страны

BigMag74 committed:
поправил навигацию

BigMag74 committed:
спрятал bottomNavigationView на экранах выбор Места работы,
выбор страны, выбор региона

BigMag74 committed:
Merge pull request #259 from anonim-no/bugfix/FLTR-258/filtering_country_navigation_fix

Bugfix/fltr 258/filtering country navigation fix

BigMag74 committed:
убрал ненужный код

manualcheg committed:
Пришлось сделать слушатель фокусировки для изменения цвета подсказки.

manualcheg committed:
Поправлен цвет endIcon

manualcheg committed:
Очищено от мусора

JudjinGM committed:
В процессе написания экрана отрасли.

JudjinGM committed:
Решение проблемы с адаптером

manualcheg committed:
Правка цвета крестика

JudjinGM committed:
1. Реализовал получение списка всех отраслей
2. Реализовал поиск по списку всех отраслей
3. Реализовал выбор единственной отрасли из списка всех отраслей и передачу во fragment result после нажатия кнопки выбрать.

JudjinGM committed:
Changes for merge

JudjinGM committed:
Merge branch 'dev' into feature/FLTR-234/industry_search

# Conflicts:
#	app/src/main/java/ru/practicum/android/diploma/common/domain/model/filter_models/AreaFilter.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/di/FilterUiModule.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/ui/AreaCountryViewHolder.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/ui/fragment/FilteringSettingsFragment.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/ui/viewModel/FilteringSettingsViewModel.kt

JudjinGM committed:
Changes for merge #2.
Исправил взаимодействие между фрагментами, поправил логику в настройках фильтров. Немного подчистил проект.

JudjinGM committed:
Merge pull request #261 from anonim-no/feature/FLTR-234/industry_search

Реализовать UI слой экрана Выбор отрасли

anonim-no committed:
- добавлен clipToPadding для RV

anonim-no committed:
- добавил обработку чекбокса "Не показывать без зп"

anonim-no committed:
- добавил клики по шевронам для перехода на экраны выбора страны и региона

anonim-no committed:
- исправлены скачки контента

anonim-no committed:
Merge branch 'dev' into debug/FLTR-255/change_color_of_hint_in_salary_edit_text

anonim-no committed:
- добавлен trim() к поисковым полям региона и отрасли

manualcheg committed:
Поправлен цвет крестика

JudjinGM committed:
Было нарушено Single Responsibility.
Разделил адаптер и диффутил на все экраны. Подчистил лишнее

JudjinGM committed:
Вынес логику datasource в репозиторий. Оказываеться в datasource не может быть несколько источников, вся логика с несколькими источниками должна быть в репозитории

JudjinGM committed:
Добавил дефолтный выбор для отрасли. Немного начал писать про дефолтный выбор региона и страны если они уже были до этого

BigMag74 committed:
Добавил отображение страны и региона, если они были в dataSource
Стираю поле региона при выборе страны
Осталось добавить автоподстановку страны при выборе региона

JudjinGM committed:
Добавил countryId в AreaFilter

JudjinGM committed:
Merge pull request #266 from anonim-no/refactoring/TECH-263/adapters

Рефакторинг приложения: свой адаптер на каждый экран.#263

BigMag74 committed:
Merge branch 'dev' of https://github.com/anonim-no/practicum-android-diploma into bugfix/FLTR-262/choosing_workplace_refactoring

JudjinGM committed:
Merge branch 'dev' into refactoring/FLTR-267/datasource

JudjinGM committed:
Merge pull request #268 from anonim-no/refactoring/FLTR-267/datasource

Рефакторинг приложения: рефакторинг datasource

JudjinGM committed:
Merge pull request #270 from anonim-no/refactoring/FLTR-269/choose_default_industry

Дефолтный выбор для отраслей

JudjinGM committed:
Fixed merge

BigMag74 committed:
Merge branch 'dev' of https://github.com/anonim-no/practicum-android-diploma into bugfix/FLTR-262/choosing_workplace_refactoring

# Conflicts:
#	app/src/main/java/ru/practicum/android/diploma/filter/data/dataSource/FiltersLocalDataSource.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/data/dataSourceImpl/FiltersLocalDataSourceImpl.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/data/repositoryImpl/FiltersRepositoryImpl.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/domain/repository/FiltersRepository.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/domain/useCase/SetAreaFilterUseCase.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/domain/useCase/SetAreaFilterUseCaseImpl.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/ui/viewModel/FilteringChoosingWorkplaceViewModel.kt

BigMag74 committed:
добавил автоподстановку страны при выборе региона

JudjinGM committed:
Поправил debounce

manualcheg committed:
первый вариант

manualcheg committed:
Второй вариант

manualcheg committed:
Merge pull request #260 from anonim-no/debug/FLTR-255/change_color_of_hint_in_salary_edit_text

Исправления поля ввода Зарплата на экране Настройки фильтрации

JudjinGM committed:
Поправил поведение кнопок.

manualcheg committed:
Merge pull request #265 from anonim-no/bugfix/SRCH-264/fix_color_of_clearing_cross

Поправлен цвет крестика

manualcheg committed:
Merge pull request #275 from anonim-no/feature/SRCH-251/start_search_after_apply_filter

Feature/srch 251/start search after apply filter

BigMag74 committed:
При стирании страны, теперь так же стираю регион

JudjinGM committed:
Добавил скролл до выбранной позиции

BigMag74 committed:
иссправил ошибку, при которой нельзя было выбрать только страну, без региона

BigMag74 committed:
Merge pull request #271 from anonim-no/bugfix/FLTR-262/choosing_workplace_refactoring

Добавил отображение страны и региона, если они были в dataSource

JudjinGM committed:
Поправил замечание. Теперь при удалении суммы кнопки уходят

JudjinGM committed:
Занулил все адаптеры итд при уничтожении вью во фрагаментах.

JudjinGM committed:
Merge pull request #274 from anonim-no/refactoring/TECH-273/debounce

Поправил debounce

JudjinGM committed:
Merge pull request #276 from anonim-no/refactoring/FLTR-272/buttons

Поправил поведение кнопок.

JudjinGM committed:
Merge branch 'dev' into refactoring/FLTR-269/choose_default_industry

# Conflicts:
#	app/src/main/java/ru/practicum/android/diploma/filter/ui/viewModel/FilteringIndustryViewModel.kt

JudjinGM committed:
Merge pull request #278 from anonim-no/refactoring/FLTR-269/choose_default_industry

Выбор отрасли. Добавил скролл до выбранной позиции отрасли

JudjinGM committed:
Fixed merge conflict.

JudjinGM committed:
Merge pull request #280 from anonim-no/refactoring/TECH-279/null-adapters

Проверить очистку и зануление адаптеров во всех фрагментах

BigMag74 committed:
Исправил ошибку перехода с экрана вакансии на экран избранного

JudjinGM committed:
Поправил баг.

JudjinGM committed:
Поправил баг. Теперь точно.

JudjinGM committed:
Merge pull request #282 from anonim-no/bugfix/SRCH-281/pagination_bugfix

Пагинация bugfix

anonim-no committed:
- отклюдчены logging-interceptor и leakcanary

JudjinGM committed:
1. Вернул обратно HeaderInterceptor
2. Поправил ошибку - вместо интерфейса передавал реализацию

JudjinGM committed:
1. Поправил нейминг и условие поиска

anonim-no committed:
- применено форматирование кода, удалены пустые строки, удален закмомментированный код, удалены неиспользуемые классы, удалены пустые директории

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/app/RootActivity.kt:
```diff
@@ -28,7 +28,8 @@ class RootActivity : AppCompatActivity() {
             when (destination.id) {
-                R.id.vacancyFragment, R.id.filteringSettingsFragment, R.id.similarVacancyFragment -> {
+                R.id.vacancyFragment, R.id.filteringSettingsFragment, R.id.similarVacancyFragment,
+                R.id.filteringChoosingWorkplaceFragment, R.id.filteringCountryFragment, R.id.filteringRegionFragment, R.id.filteringSectorFragment -> {
```
Каждый новый destination с новой строки
```suggestion
R.id.vacancyFragment, 
R.id.filteringSettingsFragment, 
R.id.similarVacancyFragment,
R.id.filteringChoosingWorkplaceFragment, 
R.id.filteringCountryFragment, 
R.id.filteringRegionFragment, 
R.id.filteringSectorFragment -> {
```

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/common/domain/model/filter_models/AreaFilter.kt:
```diff
@@ -0,0 +1,39 @@
+            return arrayOfNulls(size)
+        }
+    }
+}
```
В чем была необходимость делать класс Parcelable?

JudjinGM replied:
Объект данного класса передается из экрана выбора региона обратно на экран выбор места работы с помощью Fragment Result Api внутри bundle.
bundle.putParcelable(BUNDLE_KEY_FOR_AREA, state.areaFilter)

AleksandrIlinskii replied:
Я думаю эта статья очень сильно облегчит вам жизнь 
https://medium.com/the-lazy-coders-journal/easy-parcelable-in-kotlin-the-lazy-coders-way-9683122f4c00

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/common/domain/model/filter_models/Areas.kt:
```diff
@@ -0,0 +1,5 @@
+
+data class Areas(
+    val arealList: List<AreaFilter>,
+)
```
если 1 параметр в дата классе - его лучше делать в 1 строку

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/common/ui/mapper/VacancyDomainToVacancyUiConverter.kt:
```diff
@@ -39,53 +39,20 @@ class VacancyDomainToVacancyUiConverter(private val context: Context) {
+            salary.currency.contains("UAH") -> {context.getString(R.string.UAH)}
+            salary.currency.contains("USD") -> {context.getString(R.string.USD)}
+            salary.currency.contains("UZS") -> {context.getString(R.string.UZS)}
+            else -> ""
```
Давайте разберем этот кусок кода
Если валюта зарплаты не налл - верни пустую строку
иначе если содержит.....
иначе верни пустую строку.

Тогда в чем профит if если он в любом плохом случае вернет пустую строку?

not-a-spaceman commented on app/src/main/java/ru/practicum/android/diploma/filter/data/dataSourceImpl/FiltersLocalDataSourceImpl.kt:
```diff
@@ -0,0 +1,73 @@
+) : FiltersLocalDataSource {
+    override fun getFilterOptions(): Filter? {
+        return filterLocalCache.getFilterCache()
+    }
```
Все такие однострочные функции можно делать в 1 строку

AleksandrIlinskii replied:
Все подобные однострочные функции можно делать в 1 строку
```suggestion
    override fun getFilterOptions(): Filter? = filterLocalCache.getFilterCache()
```

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/filter/data/model/AllAreasRequest.kt:
```diff
@@ -0,0 +1,3 @@
+package ru.practicum.android.diploma.filter.data.model
+
+class AllAreasRequest
```
Зачем этот класс без ничего?

JudjinGM replied:
Данный класс создан для того, чтобы делать определенный запрос, отличный от запроса с конкретным id области в VacancyRemoteDataSourceImpl. Дело в том, что ответ, при наличии id, для получения конкретного родительского списка областей и списка всех областей отличаются. В первом случае возвращается дерево AreasDto, а во втором список деревьев List<AreaDto>. Для того, чтобы их разделить и был создан пустой dto класс AllAreasRequest.

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/filter/data/model/dto/AreasDto.kt:
```diff
@@ -0,0 +1,10 @@
+data class AreasDto(
+    val name: String,
+    val id: Int,
+    @SerializedName("parent_id") val parentId: Int?,
```
```suggestion
    @SerializedName("parent_id")
     val parentId: Int?,
```

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/filter/domain/useCase/ClearTempFilterOptionsUseCase.kt:
```diff
@@ -0,0 +1,5 @@
+
+interface ClearTempFilterOptionsUseCase {
+    fun execute()
+}
```
Это не замечание, а небольшая информация о котлине
https://kotlinlang.ru/docs/fun-interfaces.html

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/filter/domain/useCase/ClearTempFilterOptionsUseCaseImpl.kt:
```diff
@@ -0,0 +1,10 @@
+import ru.practicum.android.diploma.filter.domain.repository.FiltersRepository
+
+class ClearTempFilterOptionsUseCaseImpl(private val filtersRepository: FiltersRepository) :
+    ClearTempFilterOptionsUseCase {
```
Нужно поправить форматирование

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/filter/domain/repository/FiltersRepository.kt:
```diff
@@ -0,0 +1,38 @@
+    fun getChosenIndustry(): IndustryFilter?
+    fun getChosenArea(): AreaFilter?
+    fun getChosenCountry(): CountryFilter?
+}
```
Мне не очень нравятся такие жирные интерфейсы. Почитайте про букву I в принципе SOLID 
https://ru.wikipedia.org/wiki/SOLID_(программирование)

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/filter/ui/fragment/FilteringAreaFragment.kt:
```diff
@@ -0,0 +1,163 @@
+    private var _binding: FragmentFilteringAreaBinding? = null
+    private val binding get() = _binding!!
+    private var areasAdapter: RecycleViewAreaAdapter? = null
+    private var isClickAllowed = true
```
У вас состояние хранится в фрагменте. Для этого есть вью модель

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/filter/ui/model/FilterUi.kt:
```diff
@@ -0,0 +1,9 @@
+    val industryName: String = "",
+    val salary: String = "",
+    val onlyWithSalary: Boolean = false
+)
```
Если это класс для хранения данных - то это должен быть data class

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/filter/ui/viewModel/FilteringChoosingWorkplaceViewModel.kt:
```diff
@@ -0,0 +1,96 @@
+    val regionState: LiveData<FilterFieldsState> = _regionState
+
+    private val _selectButtonState = MutableLiveData<ButtonState>(ButtonState.Gone)
+    val selectButtonState: LiveData<ButtonState> = _selectButtonState
```
Вспомните 16 спринт.
1 лайв дата - 1 экран

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/filter/ui/viewModel/FilteringSettingsViewModel.kt:
```diff
@@ -1,6 +1,211 @@
+    private val clearAreaButtonNavigation = MutableLiveData<ClearFieldButtonNavigationState>()
+    private val clearIndustryButtonNavigation = MutableLiveData<ClearFieldButtonNavigationState>()
+    private val applyButtonState = MutableLiveData<ButtonState>()
+    private val resetButtonState = MutableLiveData<ButtonState>()
```
1 экран - 1 лайв дата

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/search/di/SearchDataModule.kt:
```diff
@@ -24,18 +20,19 @@ val searchDataModule = module {
+
+        val client = OkHttpClient.Builder()
+            .addInterceptor(headerInterceptor)
+            .build()
```
Не хватает пустой строки

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/search/ui/fragment/SearchFragment.kt:
```diff
@@ -19,44 +21,53 @@ import kotlinx.coroutines.launch
-    private var isClickAllowed = true
-
+    protected var isClickAllowed = true
+    private var isPaginationAllowed = true
```
Состояние хранится в вью модели

JudjinGM committed:
1. Убрал все liveData, объеденил все в одну liveData с общим state
2. Убрал debounce

JudjinGM committed:
Поправил логику обработки состоянии кнопки при измененных фильтров

JudjinGM committed:
1. Поправил повторно появляющийся toast

BigMag74 committed:
Изменил LiveData экрана Выбор места работы

BigMag74 committed:
Исправил LiveData экрана Выбор места работы

BigMag74 committed:
Рефакторинг ChoosingWorkplaceState

BigMag74 committed:
Vacancy LiveData

manualcheg committed:
Выделены репозитории по экранам фильтрации

JudjinGM committed:
Убрал все лишние LiveData

manualcheg committed:
Применена аннотация @Parcelize вместо реализации методов интерфейса Parcelable

anonim-no committed:
- отформатирован код в RootActivity

anonim-no committed:
- отформатирован код в Areas

anonim-no committed:
Merge pull request #299 from anonim-no/refactoring/FLTR-298/code_formatting

- отформатирован код в Areas

anonim-no committed:
- отформатирован код в AreasDTO

anonim-no committed:
Merge pull request #301 from anonim-no/refactoring/FLTR-300/code_formating_AreasDTO

- отформатирован код в AreasDTO

anonim-no committed:
- отформатирован код в ClearTempFilterOptionsUseCaseImpl

anonim-no committed:
- отформатирован код в SearchDataModule

anonim-no committed:
Merge pull request #305 from anonim-no/refactoring/SEARCH-304/code_formatting_SearchDataModule

- отформатирован код в SearchDataModule

anonim-no committed:
- FilterUi сделан data классом

BigMag74 committed:
Рефакторинг экрана выбор места работы согласно замечаниям

anonim-no committed:
- форматирование кода (однострочные функции и однострочные дата классы)

BigMag74 committed:
Рефакторинг экрана Вакансия согласно замечаниям

manualcheg committed:
Все интерфейсы репозиториев теперь содержат только используемые ими методы. Т.е. строго поделены по экранам.

manualcheg committed:
Merge pull request #293 from anonim-no/refactor/TECH-291/refactoring_repository_interfaces

10. Выделены репозитории по экранам фильтрации

BigMag74 committed:
Убрал лишний if

BigMag74 committed:
Рефакторинг после второго ревью

manualcheg committed:
Merge pull request #309 from anonim-no/refactor/FLTR-294/using_parcelize

Применена аннотация @Parcelize

BigMag74 committed:
Оставил поле isFavorite только у VacancyState.Content

BigMag74 committed:
Merge pull request #312 from anonim-no/bugfix/TECH-311/empty_string

Убрал лишний if

anonim-no committed:
Merge pull request #307 from anonim-no/bugfix/FLTR-306/FilterUi_data_class

- FilterUi сделан data классом

anonim-no committed:
Merge pull request #297 from anonim-no/refactoring/TECH-296/formating_code_in_root_activity

- отформатирован код в RootActivity

anonim-no committed:
Merge pull request #303 from anonim-no/refactoring/FLTR-302/code_formatting_ClearTempFilterOptionsUseCaseImpl

- отформатирован код в ClearTempFilterOptionsUseCaseImpl

anonim-no committed:
- правки по ревью

anonim-no committed:
- merge commit

anonim-no committed:
Merge pull request #310 from anonim-no/refactoring/TECH-308/code_formatting

- форматирование кода (однострочные функции и однострочные дата классы)

JudjinGM committed:
Поправил замечания

JudjinGM committed:
Merge pull request #287 from anonim-no/refactoring/SRCH-286/one_live-data_search_similar

13. 14. Оставить только одну LiveData на экранах поиск и похожие вакансии

JudjinGM committed:
Merge branch 'iteration_2_review' into refactoring/FLTR-289/one_live_data_filter_settings

# Conflicts:
#	app/src/main/java/ru/practicum/android/diploma/filter/ui/fragment/FilteringSettingsFragment.kt

JudjinGM committed:
Merge pull request #295 from anonim-no/refactoring/FLTR-289/one_live_data_filter_settings

13. 14. Оставить только одну LiveData на экранe настройки фильтрации

anonim-no committed:
- isClickAllowed перенесена из фрагмена во вьюмодел

anonim-no committed:
- удалены неиспользуемые импорты

anonim-no committed:
Merge pull request #315 from anonim-no/refactoring/FAV-313/clickallowed_to_viewmodel

Refactoring/fav 313/clickallowed to viewmodel

JudjinGM committed:
1. Собрал вместе фильтрацию и поиск.
2. Поправил логику при сбросе и уходе назад.

BigMag74 committed:
Поменял Serializable на Parcable у CountryFilter

JudjinGM committed:
1. Убрал лишнюю livedata.
2. Удалил лишнюю модель стейта навигации.
3. Перенес debounce во вьюмодель

JudjinGM committed:
1. Убрал лишнюю livedata.
2. Удалил лишнюю модели стейтов.
3. Перенес debounce во вьюмодель
4. Поправил SearchState, чтобы в модели было не соотояине вью,а состояние модели

BigMag74 committed:
Merge branch 'iteration_2_review' into bugfix/VACN-288/vacancy_and_choosing_workplace_liveData

BigMag74 committed:
Merge pull request #292 from anonim-no/bugfix/VACN-288/vacancy_and_choosing_workplace_liveData

13. 14. Оставить только одну LiveData на экранах Вакансия и Выбор места работы

JudjinGM committed:
Поравил ошибку, повторный поиск при ошибки после смены фильтров

JudjinGM committed:
Поправил замечание

JudjinGM committed:
Поправил замечание №2

JudjinGM committed:
Поправил замечание

JudjinGM committed:
Поправил замечание

JudjinGM committed:
Merge pull request #320 from anonim-no/refactoring/FLTR-318/one_live_data_choose_industry

13. 14. Оставить только одну LiveData на экране Выбор отрасли

JudjinGM committed:
Merge pull request #319 from anonim-no/refactoring/FLTR-317/one_live_data_choose_region

13. 14. Оставить только одну LiveData на экране Выбор региона

JudjinGM committed:
Merge pull request #316 from anonim-no/refactoring/FLTR-314/filter_search_must_work

Собрать воедино фильтрацию и поиск#314

JudjinGM committed:
1. Почистил проект
2. Удалил не использующую модель
3. Немного изменил VacancyFragment и VacancyViewModel, чтобы во фрагменте не было полей состояния

JudjinGM committed:
Убрал пустую строку

JudjinGM committed:
Изменения согласно замечаниям

JudjinGM committed:
Исправил баг. При одновременном нажатии двух полей на экранах Настройки фильтрации (место работы и Отрасль), Выбор места работы(Страна и Регион), Поиск(сразу две вакансии), был FATAL EXCEPTION (java.lang.IllegalArgumentException: Navigation action/destination). Добавил везде дебонс, а также переписал немного Выбор места работы.

anonim-no committed:
Merge pull request #322 from anonim-no/refactoring/TECH-321/final_cleaning

Финальные правки перед ревью

JudjinGM committed:
Исправил поведение при переходе обратно из похожих вакансий.
Добавил coroutineExceptionHandler в запрос через корутины.

JudjinGM committed:
1. Убрал лишние комметарии
2. Поправил наличие крестика в отрасли
3. Поправил наличие крестика в регионе

JudjinGM committed:
Поправил цвет крестика в темной теме

JudjinGM committed:
Merge branch 'refactoring/TECH-321/final_cleaning' into iteration_2_review

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/filter/data/repositoryImpl/FiltersRepositoryImpl.kt:
```diff
@@ -0,0 +1,60 @@
+
+    override fun clearTempFilterOptions() {
+        filtersLocalDataSource.clearTempFilterOptions()
+    }
```
```suggestion
    override fun clearTempFilterOptions() = filtersLocalDataSource.clearTempFilterOptions()
```

Это также можно делать в 1 строку

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/filter/ui/fragment/FilteringAreaFragment.kt:
```diff
@@ -0,0 +1,150 @@
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        recycleViewInit()
```
```suggestion
        recyclerViewInit()
```

AleksandrIlinskii replied:
Опечатка

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/filter/ui/fragment/FilteringAreaFragment.kt:
```diff
@@ -0,0 +1,150 @@
+            is AreasState.Success.Content -> showContent(state.arealList)
+            AreasState.Navigate.NavigateEmpty -> {
+                findNavController().popBackStack()
+            }
```
Выше скобки убрали, а тут зачем оставили?

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/filter/ui/fragment/FilteringChoosingWorkplaceFragment.kt:
```diff
@@ -0,0 +1,194 @@
+                getString(
+                    R.string.region
+                )
+            )
```
Тут автоформатирование немного перестаралось, это не ваша вина

anonim-no committed:
- правки по ревью
